src/analyzer.ts

```diff
@@ -154,6 +163,9 @@ export class Analyzer {
   private async _analyzeDocument(
       document: ParsedDocument<any, any>, maybeLocationOffset?: LocationOffset,
       maybeAttachedComment?: string): Promise<ScannedDocument> {
+    if (document == null) {
```

Comment:
Even in the case of an error, I'd rather not return `null`. I'd prefer returning some kind of `Result` object that has the document or errors, or incorporating errors into the document classes themselves. In general I'd like to push `null` handling upstream until we gotten rid of as much of it as we can.